chain/ethereum/src/ethereum_adapter.rs

```diff
@@ -1042,15 +1035,17 @@ where
         from: u64,
         to: u64,
         log_filter: EthereumLogFilter,
-    ) -> Box<dyn Future<Item = Vec<Log>, Error = Error> + Send> {
-        let eth = self.clone();
+    ) -> Box<dyn std::future::Future<Output = Result<Vec<Log>, Error>> + Send + Unpin> {
```

Comment:
You can use `DynTryFuture` here as well, and use `boxed` instead of `Box::new`

Reply:
Thanks, I did.